lib/blockchain.go

```diff
@@ -3539,15 +3539,15 @@ func (bc *Blockchain) CreateDAOCoinLimitOrderTxn(
 	}
 
 	// Validate txn metadata.
-	err = utxoView.IsValidDAOCoinLimitOrderMetadata(txn.PublicKey, metadata)
+	blockHeight := bc.blockTip().Height + 1
+	err = utxoView.IsValidDAOCoinLimitOrderMetadata(txn.PublicKey, blockHeight, metadata)
```

Comment:
Surfacing the new error message here doesn't require a hard fork because it's happening during transaction construction. We don't want to put this operation behind the fork height

Reply:
Good point. It's true. But IMO the benefit of reusing our existing validation functions (which run during txn construction and connection) outweighs the benefit of validating separately (without considering fork height) during txn construction in the (hopefully short) interim time between now and when we hit our fork heights.